x-pack/platform/plugins/shared/onechat/server/services/agents/persisted/client/converters.ts

```diff
@@ -74,12 +77,16 @@ export const updateRequestToEs = ({
   update: AgentUpdateRequest;
   updateDate: Date;
 }): AgentProperties => {
+  const currentConfig = currentProps.configuration ?? currentProps.config;
+
   const updated: AgentProperties = {
     ...currentProps,
     ...update,
     id: agentId,
-    configuration: {
-      ...currentProps.configuration,
+    // Explicitly omit configuration to ensure migration
+    configuration: undefined,
```

Comment:
I don't think ES _update endpoint let's you nullify field this way? I think this has to be scripted update

Also, aren't undefined values skipped when serialising json request to ES (not sure about that)